cluster-autoscaler/cloudprovider/aws/aws_manager.go

```diff
@@ -419,6 +459,18 @@ func extractLabelsFromAsg(tags []*autoscaling.TagDescription) map[string]string
 	return result
 }
 
+func extractAutoscalingOptionsFromTags(tags []*autoscaling.TagDescription) map[string]string {
+	options := make(map[string]string)
+	for _, tag := range tags {
+		splits := strings.Split(aws.StringValue(tag.Key), "k8s.io/cluster-autoscaler/node-template/autoscaling-options/")
```

Comment:
I'm not the reviewer, but this logic can occur a not expected behavior.

If the key is `[space]k8s.io/cluster-autoscaler/node-template/autoscaling-options/` , then key is `space` and value is `k8s.io/cluster-autoscaler/node-template/autoscaling-options/`.

Not a crucial problem, but that situation is not intended.

And I think this logic may be not performance critical, so we just can check the key has suffix `k8s.io/cluster-autoscaler/node-template/autoscaling-options/` and then parse the value.

This is not crucial, just a opinion.

Reply:
Note that was mirroring the way other ASG labels parsers (`extractLabelsFromAsg`, `extractAllocatableResourcesFromAsg`) are implemented, and such bogus labels are detected, ignored and logged at parsing time. I don't mind either way though, so updated to filter out on prefix as suggested.